### PR TITLE
(19w03c) Add schemas for models, tags and recipes

### PR DESCRIPTION
**Pull request by SPGoding**

### CODE REVIEW

**Comment by Levertion** [comment]
`Some descriptions in this file are from Minecraft WIKI: http://minecraft.gamepedia.com/Recipe` - Unfortunately, I don't know how well we can use these, given that we are under a  more permissive license. Obviously its incredibly unlikely that anyone would act against us, but we need to think very carefully about this.

I wonder if we can use some content from e.g. https://github.com/skylinerw/guides/blob/master/java/recipes.md instead/

File: java/data/recipe.json [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+                "ingredients": {
+                    "type": "array",
+                    "title": "A list of ingredients.",
+                    "minItems": 0,
+                    "maxItems": 9,
+                    "items": {
+                        "oneOf": [
+                            {
+                                "$ref": "#/definitions/input",
+                                "description": "An ingridient"
```
Comment (Levertion) [inline java/data/recipe.json]:
```suggestion
                                "description": "An ingredient"
```

File: java/data/recipe.json [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+                            },
+                            {
+                                "type": "array",
+                                "items": {
+                                    "$ref": "#/definitions/input"
+                                },
+                                "description": "List of possible ingridients"
```
Comment (Levertion) [inline java/data/recipe.json]:
```suggestion
                                "description": "List of possible ingredients"
```

**Comment by SPGoding** [comment]
> Unfortunately, I don't know how well we can use these, given that we are under a more permissive license. Obviously its incredibly unlikely that anyone would act against us, but we need to think very carefully about this.

I didn't notice that the two licenses are incompatible. Thank you for pointing out such a serious mistake I made!

**Comment by Levertion** [comment]
It's not that serious - its an annoying limitation however. There has been some idle talk of making a more permissive wiki, but nothing concrete has come out of it

**Pull request closed without merging**